packages/core/util/tests/util-specs.js

```diff
@@ -68,6 +122,69 @@ describe('util', () => {
 		});
 	});
 
+	describe('mergeClassNameMaps', () => {
+		const baseMap = {
+			'class-base-only': 'real-class-base-only',
+			'class-shared': 'real-class-shared-base',
+			'class-shared-another': 'real-class-shared-another-base'
+		};
+		const additiveMap = {
+			'class-shared': 'real-class-shared-additive',
+			'class-shared-another': 'real-class-shared-another-additive',
+			'class-additive-only': 'real-class-additive-only'
+		};
+
+		// Do NOT use destructuring operator as 'pxory' does not have properties but have only getters for properties.
+		const avoidProxy = (proxy) => {
+			const keys = ['class-base-only', 'class-shared', 'class-shared-another', 'class-additive-only'];
+			const obj = {};
+
+			for (let i = 0; i < keys.length; i++) {
+				if (keys[i] !== proxy[keys[i]]) {
+					obj[keys[i]] = proxy[keys[i]];
+				}
+			}
+
+			return obj;
+		};
+
+		test('should return a base map if an additive map is not given', () => {
+			const expected = {
```

Comment:
Can't this be just baseMap?

Reply:
This is intended. The interface requirement is to have the same class info, not the original reference. So, I don't make it to check the reference just to simplify the test.